src/quicktions/quicktions.pxd

```diff
@@ -0,0 +1,39 @@
+# cython: language_level=3str
+## cython: profile=True
+
+cdef extern from *:
+    """
+    #if PY_VERSION_HEX < 0x030500F0 || !CYTHON_COMPILING_IN_CPYTHON
+        #define _PyLong_GCD(a, b) (NULL)
+    #endif
+    """
+    # CPython 3.5+ has a fast PyLong GCD implementation that we can use.
+    int PY_VERSION_HEX
+    int IS_CPYTHON "CYTHON_COMPILING_IN_CPYTHON"
+    _PyLong_GCD(a, b)
+
+ctypedef unsigned long long ullong
+ctypedef unsigned long ulong
+ctypedef unsigned int uint
+
+ctypedef fused cunumber:
+    ullong
+    ulong
+    uint
+
+cpdef _gcd(a, b)
+cdef ullong _abs(long long x)
+cdef cunumber _igcd(cunumber a, cunumber b)
+cdef cunumber _ibgcd(cunumber a, cunumber b)
+cdef _py_gcd(ullong a, ullong b)
+cdef _gcd_fallback(a, b)
+
+cdef class Fraction:
+    cdef _numerator
+    cdef _denominator
+    cdef Py_hash_t _hash
```

Comment:
It's unfortunate that this attribute needs to be exposed, but that's how it is …

Reply:
It's only to create the attributes in the underlying C struct.  Those variables remain hidden from outside of Cython unless declared as `public` or `readonly`

Reply:
It's a pure implementation details that becomes visible for subclasses this way. But it's unlikely to get removed, so…